osu.Game/Overlays/Chat/DrawableChannel.cs

```diff
@@ -112,19 +137,90 @@ private void pendingMessageResolved(Message existing, Message updated)
 
         private void messageRemoved(Message removed)
         {
-            ChatLineFlow.Children.FirstOrDefault(c => c.Message == removed)?.FadeColour(Color4.Red, 400).FadeOut(600).Expire();
+            getChatLines().FirstOrDefault(c => c.Message == removed)?.FadeColour(Color4.Red, 400).FadeOut(600).Expire();
         }
 
+        private IEnumerable<ChatLine> getChatLines() => ChatLineFlow.Children.OfType<ChatLine>();
+
         private void scrollToEnd() => ScheduleAfterChildren(() => scroll.ScrollToEnd());
 
-        protected class ChatLineContainer : FillFlowContainer<ChatLine>
+        protected class ChatLineContainer : FillFlowContainer
         {
             protected override int Compare(Drawable x, Drawable y)
```

Comment:
This comparator worries me a little bit with the fallback to `base`. I'm not sure what this does, or if it's even still needed?

Reply:
Couldn't find any issues by removing this

Reply:
I've also double-checked the cases I remember being unreasonable and cannot seem to reproduce any more.